python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/api/API_IMPLEMENTATION_PLAN.md

```diff
@@ -0,0 +1,413 @@
+**GraphQL**: `repositoriesOrError`, `repositoryOrError`
+
+### 10. **workspace**
+
+```bash
+dg api workspace list [--json]
+dg api workspace view [--json]
+dg api workspace update --config <file>
+# Workspace is typically singular per deployment
+```
+
+**GraphQL**: `workspaceOrError`
+
+---
+
+## **Tier 3: Extended Management**
+
+### 11. **check**
+
+```bash
+dg api check list [--asset <asset-key>] [--status <status>] [--json]
+dg api check view <check-name> --asset <asset-key> [--json]
+# Checks are defined in code, not directly manageable
+```
+
+**GraphQL**: `assetCheckExecutions`
+
+### 12. **schedule**
+
+```bash
+dg api schedule list [--location <name>] [--status <running|stopped>] [--json]
+dg api schedule view <schedule-name> --location <name> [--json]
+dg api schedule start <schedule-name> --location <name>
+dg api schedule stop <schedule-name> --location <name>
+```
+
+**GraphQL**: `schedulesOrError`, `scheduleOrError`
+
+### 13. **sensor**
+
+```bash
+dg api sensor list [--location <name>] [--status <running|stopped>] [--json]
+dg api sensor view <sensor-name> --location <name> [--json]
+dg api sensor start <sensor-name> --location <name>
+dg api sensor stop <sensor-name> --location <name>
+```
+
+**GraphQL**: `sensorsOrError`, `sensorOrError`
+
+### 14. **backfill**
+
+```bash
+dg api backfill list [--status <status>] [--json]
+dg api backfill view <backfill-id> [--json]
+dg api backfill create --asset <asset-key> --partitions <range>
+dg api backfill cancel <backfill-id>
+```
+
+**GraphQL**: `partitionBackfillsOrError`, `partitionBackfillOrError`
+
+### 15. **env-var**
+
+```bash
+dg api env-var list [--location <name>] [--json]
+dg api env-var view <var-name> [--location <name>] [--json]
+# Environment variables managed through deployment settings
+```
+
+**GraphQL**: `utilizedEnvVarsOrError`
+
+---
+
+## **Tier 4: Advanced/Monitoring**
+
+### 16. **alert-notification**
+
+```bash
+dg api alert-notification list [--policy-id <id>] [--limit <n>] [--json]
+dg api alert-notification view <notification-id> [--json]
+# Notifications are read-only audit trail
+```
+
+**GraphQL**: `alertPolicyNotifications`
+
+### 17. **audit-log**
+
+```bash
+dg api audit-log list [--user <user-id>] [--action <action>] [--limit <n>] [--json]
+dg api audit-log view <log-id> [--json]
+# Audit logs are read-only
+```
+
+**GraphQL**: `auditLog`
+
+### 18. **custom-role**
+
+```bash
+dg api custom-role list [--json]
+dg api custom-role view <role-id> [--json]
+dg api custom-role create --name <name> --permissions <file>
+dg api custom-role update <role-id> --permissions <file>
+dg api custom-role delete <role-id>
+```
+
+**GraphQL**: `customRoles`, `customRoleOrError`
+
+### 19. **api-token**
+
+```bash
+dg api api-token list [--type <user|agent>] [--json]
+dg api api-token view <token-id> [--json]
+dg api api-token create --description <desc> [--type <type>]
+dg api api-token revoke <token-id>
+```
+
+**GraphQL**: `apiTokensOrError`, `agentTokensOrError`
```

Comment:
confusingly api tokens and agent tokens are different things

maybe just stick to agent tokens for now